### ISSUE TITLE
Issue with accessing MegaDetector weight file

### ISSUE THREAD

**Issue by ronbitonn** [agentmorris/MegaDetector#102]
I am experiencing difficulties accessing the MegaDetector weight file when running the following command in Google Colab notebook:

ruby
Copy code
!wget https://lilablobssc.blob.core.windows.net/models/camera_traps/megadetector/md_v4.1.0/md_v4.1.0.pb
Previously, this command worked fine, allowing me to download the MegaDetector weight file (md_v4.1.0.pb) from the specified URL. However, I now receive the following error message:

vbnet
Copy code
--2023-07-03 09:36:45--  https://lilablobssc.blob.core.windows.net/models/camera_traps/megadetector/md_v4.1.0/md_v4.1.0.pb
Resolving lilablobssc.blob.core.windows.net (lilablobssc.blob.core.windows.net)... 20.60.149.97
Connecting to lilablobssc.blob.core.windows.net (lilablobssc.blob.core.windows.net)|20.60.149.97|:443... connected.
HTTP request sent, awaiting response... 409 Public access is not permitted on this storage account.
2023-07-03 09:36:45 ERROR 409: Public access is not permitted on this storage account.
The error indicates that public access is not allowed on the storage account hosting the MegaDetector weight file. I tried this command around a month ago, and it worked without any issues.

I kindly request assistance in resolving this issue. Is there an alternative source where I can access the MegaDetector weight file (md_v4.1.0.pb), or could you provide guidance on how to obtain it?

If needed, I can share my Google Colab notebook via email for further investigation.

Thank you for your support!

**Comment by agentmorris** [comment]
Sorry about that; that was a temporary issue with the lilablobssc storage account. It has since been fixed, so your wget command should work fine now.

But, now you've got me intrigued... totally out of curiosity, can you share some more about why you're using MDv4?  It's not that there's *never* a time to use MDv4, it's just relatively uncommon, so I'm curious when it comes up.

**Comment by agentmorris** [comment]
Also, I saw you posted a related issue on the Microsoft/CameraTraps issue, about PyTorch versions... FWIW, if you are in fact using MDv4, you can skip PyTorch installation entirely (MDv4 uses TensorFlow, MDv5 uses PyTorch).  But, assuming you want to use MDv5 in Colab, the issue you encountered has already been fixed in the version of the [notebook](https://github.com/agentmorris/MegaDetector/blob/main/detection/megadetector_colab.ipynb) on this repo (agentmorris/MegaDetector).

For posterity, the issue is that MDv5 depends on a slightly old version of PyTorch, and a few months ago, old versions of PyTorch were removed from the standard pip distribution channels.  The notebook was updated to use a slightly different build of MDv5 that is compatible with newer versions of PyTorch.

**Comment by ronbitonn** [comment]
Thank you for your response and assistance, Agent Morris. I'm using MegaDetector (MD) for animal detection in my MSc thesis research. Initially, I used MDv4 for experiments with sonar data and found it to work well. I'm currently trying it on Google Colab for proof of concept. I encountered issues with MDv5 and PyTorch versions, which I posted on the Microsoft/CameraTraps repo. Once I confirm MD's functionality on Colab, I plan to use it on my Jetson Orin AGX machine (arm64). I would appreciate guidance on setting it up for that platform. Thank you for your support.

Best regards,
Ron.

**Comment by ronbitonn** [comment]
Thank you for the update! I'm glad to hear that the updated Colab notebook is working well now. It's true that MDv5 offers improved performance and accuracy compared to MDv4, so it's worth considering for future work.

Regarding running MegaDetector on Jetson hardware, referring to the thread you mentioned is a great idea. Luke, who has experience running MDv5 on Jetson devices, could provide valuable insights and guidance. 

 Thanks and have a wonderful day!
